Expo SDK 50 Compatibility/Upgrade

[Expo SDK 50 has released](https://expo.dev/changelog/2024/01-18-sdk-50)

It contains the following:
> [Notable breaking changes](https://expo.dev/changelog/2024/01-18-sdk-50#notable-breaking-changes)
>
> - Android SDK 34, AGP 8, and Java 17. If you build your project locally, you will need to install JDK 17. [Learn more](https://docs.expo.dev/guides/local-app-development/#android).
> - Android minimum supported version bumped to Android 6 (API 23).

Android SDK/Minimum version bump seems to be in-line with the required versions in the module README.md, I'd assume that would mean that the build-properties is no longer required. The module must also upgrade to Java 17 for compatibility.

Tried a completely fresh expo project with this module earlier today (having some problems with a crash on startup for production builds, which from the testing I did doesn't seem directly related to this module).

Anyways the `npx eas-cli build --platform android` command ended up failing with the following exception:

```
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':youssefhenna-expo-enode-link-sdk:compileReleaseKotlin'.
> 'compileReleaseJavaWithJavac' task (current target is 17) and 'compileReleaseKotlin' task (current target is 11) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain
```

I did try to just edit the mentions of `JavaVersion.VERSION_11` in the build.gradle directly in the node_modules directory, but didn't seem to help. Might've missed something.

<details closed>
<summary>Full logs</summary>

```
Running 'gradlew :app:bundleRelease' in /home/expo/workingdir/build/android
Downloading https://services.gradle.org/distributions/gradle-8.3-all.zip
10%
20%.
30%.
40%
50%
60%.
70%.
80%.
90%.
100%
Welcome to Gradle 8.3!
Here are the highlights of this release:
 - Faster Java compilation
 - Reduced memory usage
- Support for running on Java 20
For more details see https://docs.gradle.org/8.3/release-notes.html
To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/8.3/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
Daemon will be stopped at the end of the build
> Configure project :
Path for java installation '/usr/lib/jvm/openjdk-17' (Common Linux Locations) does not contain a java executable
> Task :gradle-plugin:pluginDescriptors
> Task :gradle-plugin:processResources
> Task :gradle-plugin:compileKotlin
> Task :gradle-plugin:compileJava
NO-SOURCE
> Task :gradle-plugin:classes
> Task :gradle-plugin:jar
> Task :gradle-plugin:inspectClassesForKotlinIC
> Configure project :expo-modules-core
Checking the license for package NDK (Side by side) 25.1.8937393 in /home/expo/Android/Sdk/licenses
License for package NDK (Side by side) 25.1.8937393 accepted.
Preparing "Install NDK (Side by side) 25.1.8937393 v.25.1.8937393".
"Install NDK (Side by side) 25.1.8937393 v.25.1.8937393" ready.
Installing NDK (Side by side) 25.1.8937393 in /home/expo/Android/Sdk/ndk/25.1.8937393
"Install NDK (Side by side) 25.1.8937393 v.25.1.8937393" complete.
"Install NDK (Side by side) 25.1.8937393 v.25.1.8937393" finished.
> Configure project :expo-modules-core$android-annotation
Path for java installation '/usr/lib/jvm/openjdk-17' (Common Linux Locations) does not contain a java executable
> Configure project :expo
Using expo modules
  - [32myoussefhenna-expo-enode-link-sdk[0m (0.3.5)
  - [32mexpo-constants[0m (15.4.5)
  - [32mexpo-file-system[0m (16.0.5)
  - [32mexpo-font[0m (11.10.2)
  - [32mexpo-keep-awake[0m (12.8.2)
  - [32mexpo-modules-core[0m (1.11.8)
  - [32mexpo-modules-core$android-annotation[0m (1.11.8)
  - [32mexpo-modules-core$android-annotation-processor[0m (1.11.8)
Checking the license for package Android SDK Build-Tools 34 in /home/expo/Android/Sdk/licenses
License for package Android SDK Build-Tools 34 accepted.
Preparing "Install Android SDK Build-Tools 34 v.34.0.0".
"Install Android SDK Build-Tools 34 v.34.0.0" ready.
Installing Android SDK Build-Tools 34 in /home/expo/Android/Sdk/build-tools/34.0.0
"Install Android SDK Build-Tools 34 v.34.0.0" complete.
"Install Android SDK Build-Tools 34 v.34.0.0" finished.
Checking the license for package Android SDK Platform 34 in /home/expo/Android/Sdk/licenses
License for package Android SDK Platform 34 accepted.
Preparing "Install Android SDK Platform 34 (revision 2)".
"Install Android SDK Platform 34 (revision 2)" ready.
Installing Android SDK Platform 34 in /home/expo/Android/Sdk/platforms/android-34
"Install Android SDK Platform 34 (revision 2)" complete.
"Install Android SDK Platform 34 (revision 2)" finished.
Checking the license for package Android SDK Build-Tools 33.0.1 in /home/expo/Android/Sdk/licenses
License for package Android SDK Build-Tools 33.0.1 accepted.
Preparing "Install Android SDK Build-Tools 33.0.1 v.33.0.1".
"Install Android SDK Build-Tools 33.0.1 v.33.0.1" ready.
Installing Android SDK Build-Tools 33.0.1 in /home/expo/Android/Sdk/build-tools/33.0.1
"Install Android SDK Build-Tools 33.0.1 v.33.0.1" complete.
"Install Android SDK Build-Tools 33.0.1 v.33.0.1" finished.
> Task :expo-file-system:preBuild UP-TO-DATE
> Task :expo-file-system:preReleaseBuild UP-TO-DATE
> Task :expo-file-system:generateReleaseResValues
> Task :expo-file-system:generateReleaseResources
> Task :expo-file-system:packageReleaseResources
> Task :expo-font:preBuild UP-TO-DATE
> Task :expo-font:preReleaseBuild UP-TO-DATE
> Task :expo-font:generateReleaseResValues
> Task :expo-font:generateReleaseResources
> Task :expo-font:packageReleaseResources
> Task :expo-keep-awake:preBuild UP-TO-DATE
> Task :expo-keep-awake:preReleaseBuild UP-TO-DATE
> Task :expo-keep-awake:generateReleaseResValues
> Task :expo-keep-awake:generateReleaseResources
> Task :expo-keep-awake:packageReleaseResources
> Task :expo-modules-core:preBuild UP-TO-DATE
> Task :expo-modules-core:preReleaseBuild UP-TO-DATE
> Task :expo-modules-core:generateReleaseResValues
> Task :expo-modules-core:generateReleaseResources
> Task :expo-constants:createExpoConfig
> Task :expo:generateExpoModulesPackageList
> Task :expo-constants:preBuild
> Task :expo:preBuild
> Task :expo-constants:preReleaseBuild
> Task :expo:preReleaseBuild
> Task :expo:generateReleaseResValues
> Task :expo:generateReleaseResources
> Task :expo-constants:generateReleaseResValues
> Task :expo-constants:generateReleaseResources
> Task :expo-modules-core:packageReleaseResources
> Task :youssefhenna-expo-enode-link-sdk:preBuild UP-TO-DATE
> Task :youssefhenna-expo-enode-link-sdk:preReleaseBuild UP-TO-DATE
> Task :youssefhenna-expo-enode-link-sdk:generateReleaseResValues
> Task :youssefhenna-expo-enode-link-sdk:generateReleaseResources
> Task :expo-constants:packageReleaseResources
> Task :expo:packageReleaseResources
> Task :expo-constants:extractDeepLinksRelease
> Task :expo:extractDeepLinksRelease
> Task :youssefhenna-expo-enode-link-sdk:packageReleaseResources
> Task :expo-file-system:extractDeepLinksRelease
> Task :expo-file-system:processReleaseManifest
/home/expo/workingdir/build/node_modules/expo-file-system/android/src/main/AndroidManifest.xml:6:9-8:20 Warning:
	provider#expo.modules.filesystem.FileSystemFileProvider@android:authorities was tagged at AndroidManifest.xml:6 to replace other declarations but no other declaration present
> Task :expo:processReleaseManifest
> Task :expo-keep-awake:extractDeepLinksRelease
> Task :expo-font:extractDeepLinksRelease
> Task :expo-constants:processReleaseManifest
> Task :expo-modules-core:extractDeepLinksRelease
> Task :expo-font:processReleaseManifest
> Task :expo-keep-awake:processReleaseManifest
> Task :youssefhenna-expo-enode-link-sdk:extractDeepLinksRelease
> Task :expo-modules-core:processReleaseManifest
/home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/AndroidManifest.xml:8:9-11:45 Warning:
	meta-data#com.facebook.soloader.enabled@android:value was tagged at AndroidManifest.xml:8 to replace other declarations but no other declaration present
> Task :youssefhenna-expo-enode-link-sdk:processReleaseManifest
> Task :expo:compileReleaseLibraryResources
> Task :expo-constants:compileReleaseLibraryResources
> Task :expo-file-system:compileReleaseLibraryResources
> Task :expo-font:compileReleaseLibraryResources
> Task :expo-keep-awake:compileReleaseLibraryResources
> Task :expo-modules-core:compileReleaseLibraryResources
> Task :youssefhenna-expo-enode-link-sdk:compileReleaseLibraryResources
> Task :expo:generateReleaseBuildConfig
> Task :expo-modules-core$android-annotation-processor:processResources
> Task :expo-constants:generateReleaseBuildConfig
> Task :expo-constants:parseReleaseLocalResources
> Task :expo:parseReleaseLocalResources
> Task :expo:generateReleaseRFile
> Task :expo-constants:generateReleaseRFile
> Task :expo-modules-core:generateReleaseBuildConfig
> Task :expo-constants:javaPreCompileRelease
> Task :expo-file-system:generateReleaseBuildConfig
> Task :expo-modules-core:parseReleaseLocalResources
> Task :expo-modules-core:javaPreCompileRelease
> Task :expo-modules-core:generateReleaseRFile
> Task :expo-file-system:javaPreCompileRelease
> Task :expo-font:generateReleaseBuildConfig
> Task :expo-font:parseReleaseLocalResources
> Task :expo-font:generateReleaseRFile
> Task :expo-file-system:parseReleaseLocalResources
> Task :expo-font:javaPreCompileRelease
> Task :expo-keep-awake:generateReleaseBuildConfig
> Task :expo-file-system:generateReleaseRFile
> Task :youssefhenna-expo-enode-link-sdk:generateReleaseBuildConfig
> Task :expo-keep-awake:parseReleaseLocalResources
> Task :youssefhenna-expo-enode-link-sdk:parseReleaseLocalResources
> Task :expo-keep-awake:generateReleaseRFile
> Task :expo-keep-awake:javaPreCompileRelease
> Task :expo:javaPreCompileRelease
> Task :youssefhenna-expo-enode-link-sdk:generateReleaseRFile
> Task :youssefhenna-expo-enode-link-sdk:javaPreCompileRelease
> Task :expo:writeReleaseAarMetadata
> Task :expo-constants:writeReleaseAarMetadata
> Task :expo-file-system:writeReleaseAarMetadata
> Task :expo-font:writeReleaseAarMetadata
> Task :expo-keep-awake:writeReleaseAarMetadata
> Task :expo-modules-core:writeReleaseAarMetadata
> Task :youssefhenna-expo-enode-link-sdk:writeReleaseAarMetadata
> Task :expo:mergeReleaseShaders
> Task :expo:compileReleaseShaders NO-SOURCE
> Task :expo:generateReleaseAssets UP-TO-DATE
> Task :expo:packageReleaseAssets
> Task :expo-constants:mergeReleaseShaders
> Task :expo-constants:compileReleaseShaders NO-SOURCE
> Task :expo-constants:generateReleaseAssets UP-TO-DATE
> Task :expo-constants:packageReleaseAssets
> Task :expo-modules-core$android-annotation:processResources NO-SOURCE
> Task :expo-file-system:mergeReleaseShaders
> Task :expo-file-system:compileReleaseShaders NO-SOURCE
> Task :expo-file-system:generateReleaseAssets UP-TO-DATE
> Task :expo-file-system:packageReleaseAssets
> Task :expo-font:mergeReleaseShaders
> Task :expo-font:compileReleaseShaders NO-SOURCE
> Task :expo-font:generateReleaseAssets UP-TO-DATE
> Task :expo-font:packageReleaseAssets
> Task :expo-keep-awake:mergeReleaseShaders
> Task :expo-keep-awake:compileReleaseShaders NO-SOURCE
> Task :expo-keep-awake:generateReleaseAssets UP-TO-DATE
> Task :expo-keep-awake:packageReleaseAssets
> Task :expo-modules-core:mergeReleaseShaders
> Task :expo-modules-core:compileReleaseShaders NO-SOURCE
> Task :expo-modules-core:generateReleaseAssets UP-TO-DATE
> Task :expo-modules-core:packageReleaseAssets
> Task :youssefhenna-expo-enode-link-sdk:mergeReleaseShaders
> Task :youssefhenna-expo-enode-link-sdk:compileReleaseShaders NO-SOURCE
> Task :youssefhenna-expo-enode-link-sdk:generateReleaseAssets UP-TO-DATE
> Task :youssefhenna-expo-enode-link-sdk:packageReleaseAssets
> Task :expo:mergeReleaseJniLibFolders
> Task :expo:mergeReleaseNativeLibs NO-SOURCE
> Task :expo:copyReleaseJniLibsProjectOnly
> Task :expo-constants:mergeReleaseJniLibFolders
> Task :expo-constants:mergeReleaseNativeLibs NO-SOURCE
> Task :expo-constants:copyReleaseJniLibsProjectOnly
> Task :expo-file-system:mergeReleaseJniLibFolders
> Task :expo-file-system:mergeReleaseNativeLibs NO-SOURCE
> Task :expo-file-system:copyReleaseJniLibsProjectOnly
> Task :expo-font:mergeReleaseJniLibFolders
> Task :expo-font:mergeReleaseNativeLibs NO-SOURCE
> Task :expo-font:copyReleaseJniLibsProjectOnly
> Task :expo-keep-awake:mergeReleaseJniLibFolders
> Task :expo-keep-awake:mergeReleaseNativeLibs NO-SOURCE
> Task :expo-keep-awake:copyReleaseJniLibsProjectOnly
> Task :expo-modules-core$android-annotation:compileKotlin
> Task :expo-modules-core$android-annotation:compileJava NO-SOURCE
> Task :expo-modules-core$android-annotation:classes UP-TO-DATE
> Task :expo-modules-core$android-annotation:jar
> Task :expo-modules-core$android-annotation:inspectClassesForKotlinIC
> Task :expo-modules-core:configureCMakeRelWithDebInfo[arm64-v8a]
Checking the license for package CMake 3.22.1 in /home/expo/Android/Sdk/licenses
License for package CMake 3.22.1 accepted.
Preparing "Install CMake 3.22.1 v.3.22.1".
"Install CMake 3.22.1 v.3.22.1" ready.
Installing CMake 3.22.1 in /home/expo/Android/Sdk/cmake/3.22.1
"Install CMake 3.22.1 v.3.22.1" complete.
"Install CMake 3.22.1 v.3.22.1" finished.
> Task :expo-modules-core$android-annotation-processor:compileKotlin
> Task :expo-modules-core$android-annotation-processor:compileJava NO-SOURCE
> Task :expo-modules-core$android-annotation-processor:classes
> Task :expo-modules-core$android-annotation-processor:jar
> Task :expo-modules-core$android-annotation-processor:inspectClassesForKotlinIC
> Task :expo:prepareReleaseArtProfile
> Task :expo-constants:prepareReleaseArtProfile
> Task :expo-file-system:prepareReleaseArtProfile
> Task :expo-font:prepareReleaseArtProfile
> Task :expo-keep-awake:prepareReleaseArtProfile
> Task :youssefhenna-expo-enode-link-sdk:mergeReleaseJniLibFolders
> Task :youssefhenna-expo-enode-link-sdk:mergeReleaseNativeLibs NO-SOURCE
> Task :youssefhenna-expo-enode-link-sdk:prepareReleaseArtProfile
> Task :youssefhenna-expo-enode-link-sdk:copyReleaseJniLibsProjectOnly
> Task :app:createBundleReleaseJsAndAssets
Writing bundle output to: /home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle
Writing sourcemap output to: /home/expo/workingdir/build/android/app/build/intermediates/sourcemaps/react/release/index.android.bundle.packager.map
Done writing bundle output
Done writing sourcemap output
> Task :expo-modules-core:compileReleaseKotlin
> Task :expo-modules-core:buildCMakeRelWithDebInfo[arm64-v8a]
> Task :app:createBundleReleaseJsAndAssets
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:3485:16: warning: the variable "SharedArrayBuffer" was not declared in function "from"
if (typeof SharedArrayBuffer !== 'undefined' && (isInstance(value, Shared...
^~~~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle
:
8262
:18: warning: the variable "DebuggerInternal" was not declared in function "__shouldPauseOnThrow"
          typeof DebuggerInternal !== 'undefined' &&
                 ^~~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:19040:7: warning: the variable "setTimeout" was not declared in function "logCapturedError"
      setTimeout(function () {
      ^~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:17000
:108: warning: the variable "nativeFabricUIManager" was not declared in function "onChange"
...lInstanceHandle ? (from && nativeFabricUIManager.setIsJSResponder(from.sta...
                              ^~~~~~~~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:17621:21: warning: the variable "clearTimeout" was not declared in anonymous function " 345#"
    cancelTimeout = clearTimeout;
                    ^~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:21371:30: warning: the variable "__REACT_DEVTOOLS_GLOBAL_HOOK__" was not declared in anonymous function " 345#"
  if ("undefined" !== typeof __REACT_DEVTOOLS_GLOBAL_HOOK__) {
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:22903:5: warning: the variable "setImmediate" was not declared in function "handleResolved"
    setImmediate(function () {
    ^~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:26913:5: warning: the variable "fetch" was not declared in anonymous function " 558#"
    fetch,
    ^~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:26914:5: warning: the variable "Headers" was not declared in anonymous function " 558#"
    Headers,
    ^~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:26915:5: warning: the variable "Request" was not declared in anonymous function " 558#"
    Request,
    ^~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:26916:5: warning: the variable "Response" was not declared in anonymous function " 558#"
    Response
    ^~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:27073:24: warning: the variable "FileReader" was not declared in function "readBlobAsArrayBuffer"
      var reader = new FileReader();
                       ^~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:27124:36: warning: the variable "Blob" was not declared in anonymous function " 569#"
        } else if (support.blob && Blob.prototype.isPrototypeOf(body)) {
                                   ^~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:27126:40: warning: the variable "FormData" was not declared in anonymous function " 569#"
        } else if (support.formData && FormData.prototype.isPrototypeOf(body)) {
                                       ^~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:27128:44: warning: the variable "URLSearchParams" was not declared in anonymous function " 569#"
...e if (support.searchParams && URLSearchParams.prototype.isPrototypeOf(body...
                                 ^~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:27247:26: warning: the variable "AbortController" was not declared in anonymous function " 575#"
          var ctrl = new AbortController();
                         ^~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:27381:23: warning: the variable "XMLHttpRequest" was not declared in anonymous function " 579#"
        var xhr = new XMLHttpRequest();
                      ^~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:26926:71: warning: the variable "self" was not declared in anonymous function " 561#"
...undefined' && globalThis || typeof self !== 'undefined' && self ||
                                      ^~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:31616:27: warning: the variable "performance" was not declared in anonymous function " 753#"
  if ("object" === typeof performance && "function" === typeof performance.no...
                          ^~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:31639:26: warning: the variable "navigator" was not declared in anonymous function " 753#"
  "undefined" !== typeof navigator && undefined !== navigator.scheduling && u...
                         ^~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:31749:37: warning: the variable "MessageChannel" was not declared in anonymous function " 753#"
  };else if ("undefined" !== typeof MessageChannel) {
                                    ^~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:31764:34: warning: the variable "nativeRuntimeScheduler" was not declared in anonymous function " 753#"
... = "undefined" !== typeof nativeRuntimeScheduler ? nativeRuntimeScheduler....
                             ^~~~~~~~~~~~~~~~~~~~~~
/home/expo/workingdir/build/android/app/build/generated/assets/createBundleReleaseJsAndAssets/index.android.bundle:43720:34: warning: the variable "requestAnimationFrame" was not declared in function "start 9#"
...    this._animationFrame = requestAnimationFrame(this.onUpdate.bind(this));
                              ^~~~~~~~~~~~~~~~~~~~~
> Task :app:generateCodegenSchemaFromJavaScript SKIPPED
> Task :app:generateCodegenArtifactsFromSchema SKIPPED
> Task :app:generatePackageList
> Task :app:preBuild
> Task :app:preReleaseBuild
> Task :app:generateReleaseResValues
> Task :expo-modules-core:configureCMakeRelWithDebInfo[armeabi-v7a]
> Task :app:mapReleaseSourceSetPaths
> Task :app:generateReleaseResources
> Task :app:createReleaseCompatibleScreenManifests
> Task :app:extractDeepLinksRelease
> Task :app:processReleaseMainManifest
/home/expo/workingdir/build/android/app/src/main/AndroidManifest.xml Warning:
	provider#expo.modules.filesystem.FileSystemFileProvider@android:authorities was tagged at AndroidManifest.xml:0 to replace other declarations but no other declaration present
> Task :app:processReleaseManifest
> Task :app:processApplicationManifestReleaseForBundle
> Task :app:mergeReleaseResources
> Task :app:checkReleaseDuplicateClasses
> Task :app:buildKotlinToolingMetadata
> Task :app:generateReleaseBuildConfig
> Task :app:checkReleaseAarMetadata
> Task :app:packageReleaseResources
> Task :app:parseReleaseLocalResources
> Task :app:processReleaseManifestForPackage
> Task :expo-modules-core:buildCMakeRelWithDebInfo[armeabi-v7a]
> Task :app:bundleReleaseResources
> Task :app:javaPreCompileRelease
> Task :app:processReleaseResources
> Task :app:desugarReleaseFileDependencies
> Task :expo-modules-core:configureCMakeRelWithDebInfo[x86]
> Task :app:mergeExtDexRelease
> Task :expo-modules-core:buildCMakeRelWithDebInfo[x86]
> Task :expo-modules-core:compileReleaseKotlin
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/core/utilities/EmulatorUtilities.kt:30:13 'SERIAL: String!' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:151:13 Variable 'catalystInstance' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/ExpoModulesHelper.kt:11:21 'newInstance(): T!' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/Promise.kt:72:18 This declaration overrides deprecated member but not marked as deprecated itself. This deprecation won't be inherited in future releases. Please add @Deprecated annotation or suppress. See https://youtrack.jetbrains.com/issue/KT-47902 for details
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/Utils.kt:8:3 Expected performance impact from inlining is insignificant. Inlining works best for functions with parameters of functional types
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/activityresult/ActivityResultsManager.kt:51:24 Parameter 'activity' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/activityresult/AppContextActivityResultRegistry.kt:119:51 'getParcelableExtra(String!): T?' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/activityresult/AppContextActivityResultRegistry.kt:186:26 'getParcelable(String?): T?' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/activityresult/AppContextActivityResultRegistry.kt:277:83 'getParcelable(String?): T?' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/activityresult/DataPersistor.kt:67:20 'getSerializable(String?): Serializable?' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/activityresult/DataPersistor.kt:85:26 'getSerializable(String?): Serializable?' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/devtools/cdp/CdpNetworkTypes.kt:186:54 Parameter 'request' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/devtools/cdp/CdpNetworkTypes.kt:210:54 Parameter 'request' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/devtools/cdp/CdpNetworkTypes.kt:230:15 Parameter 'now' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/devtools/cdp/CdpNetworkTypes.kt:230:54 Parameter 'request' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/events/KModuleEventEmitterWrapper.kt:90:7 'constructor Event<T : Event<(raw) Event<*>>!>(Int)' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptObject.kt:90:33 Parameter 'null' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptObject.kt:91:34 Parameter 'null' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/jni/JavaScriptObject.kt:132:5 Parameter 'null' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/sharedobjects/SharedObjectRegistry.kt:54:35 Destructured parameter 'js' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/tracing/ExpoTrace.kt:33:1 Expected performance impact from inlining is insignificant. Inlining works best for functions with parameters of functional types
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/tracing/ExpoTrace.kt:40:1 Expected performance impact from inlining is insignificant. Inlining works best for functions with parameters of functional types
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/tracing/ExpoTrace.kt:50:1 Expected performance impact from inlining is insignificant. Inlining works best for functions with parameters of functional types
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/tracing/ExpoTrace.kt:56:1 Expected performance impact from inlining is insignificant. Inlining works best for functions with parameters of functional types
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:76:12 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:79:12 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:82:11 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:85:11 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:99:12 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:102:11 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:114:12 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/Either.kt:117:11 Parameter 'type' is never used
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/JSTypeConverterHelper.kt:44:17 'get(String!): Any?' is deprecated. Deprecated in Java
w: file:///home/expo/workingdir/build/node_modules/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/TypeConverterProvider.kt:175:46 'newInstance(): T!' is deprecated. Deprecated in Java
> Task :expo-modules-core:compileReleaseJavaWithJavac
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
> Task :expo-modules-core:bundleLibCompileToJarRelease
> Task :expo-constants:compileReleaseKotlin
w: file:///home/expo/workingdir/build/node_modules/expo-constants/android/src/main/java/expo/modules/constants/ConstantsService.kt:127:14 'versionCode: Int' is deprecated. Deprecated in Java
> Task :expo-constants:compileReleaseJavaWithJavac
> Task :expo-modules-core:bundleLibRuntimeToJarRelease
> Task :expo-font:compileReleaseKotlin
> Task :expo-font:compileReleaseJavaWithJavac
> Task :expo-font:bundleLibCompileToJarRelease
> Task :expo-keep-awake:compileReleaseKotlin
> Task :expo-constants:bundleLibCompileToJarRelease
> Task :expo-keep-awake:compileReleaseJavaWithJavac
> Task :expo-keep-awake:bundleLibCompileToJarRelease
> Task :youssefhenna-expo-enode-link-sdk:compileReleaseKotlin FAILED
> Task :expo-constants:bundleLibRuntimeToJarRelease
> Task :expo-file-system:compileReleaseKotlin
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':youssefhenna-expo-enode-link-sdk:compileReleaseKotlin'.
> 'compileReleaseJavaWithJavac' task (current target is 17) and 'compileReleaseKotlin' task (current target is 11) jvm target compatibility should be set to the same Java version.
  Consider using JVM toolchain: https://kotl.in/gradle/jvm/toolchain
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
BUILD FAILED in 5m 44s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
For more on this, please refer to https://docs.gradle.org/8.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
160 actionable tasks: 160 executed
Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.
```
</details>

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -53,13 +53,16 @@ afterEvaluate {
 android {
   compileSdkVersion safeExtGet("compileSdkVersion", 34)
 
-  compileOptions {
-    sourceCompatibility JavaVersion.VERSION_11
-    targetCompatibility JavaVersion.VERSION_11
-  }
+  def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+  if (agpVersion.tokenize('.')[0].toInteger() < 8) {
+    compileOptions {
+      sourceCompatibility JavaVersion.VERSION_11
+      targetCompatibility JavaVersion.VERSION_11
+    }
 
-  kotlinOptions {
-    jvmTarget = JavaVersion.VERSION_11.majorVersion
+    kotlinOptions {
+      jvmTarget = JavaVersion.VERSION_11.majorVersion
+    }
   }
 
   namespace "expo.modules.enodelinksdk"

--- a/ios/ExpoEnodeLinkSDK.podspec
+++ b/ios/ExpoEnodeLinkSDK.podspec
@@ -10,7 +10,7 @@ Pod::Spec.new do |s|
   s.license        = package['license']
   s.author         = package['author']
   s.homepage       = package['homepage']
-  s.platform       = :ios, '13.0'
+  s.platform       = :ios, '13.4'
   s.swift_version  = '5.4'
   s.source         = { git: 'https://github.com/YoussefHenna/expo-enode-link-sdk' }
   s.static_framework = true

